mavros_extras/src/plugins/esc_telemetry.cpp

```diff
@@ -21,6 +25,60 @@ namespace mavros
 {
 namespace extra_plugins
 {
+// for all the dignostics parameters
+struct Limits {
+	bool _diag_enabled;
+	// Temperature checks
+	float _temp_min_nok;
+	float _temp_min_ok;
+	float _temp_max_nok;
+	float _temp_max_ok;
+	// Voltage checks
+	float _volt_min_nok;
+	float _volt_min_ok;
+	float _volt_max_nok;
+	float _volt_max_ok;
+	// Current checks
+	float _curr_min_nok;
+	float _curr_min_ok;
+	float _curr_max_nok;
+	float _curr_max_ok;
+	// RPM checks
+	int _rpm_min_nok;
+	int _rpm_min_ok;
+	int _rpm_max_nok;
+	int _rpm_max_ok;
+	// Count checks, should be uint16_t but ROS only allows int parameters
+	int _count_min_delta;
+
+	explicit Limits(ros::NodeHandle &pnh) {
+		if (!pnh.getParam("enabled", this->_diag_enabled)) {
+			ROS_FATAL_STREAM("Could not retrieve enabled parameter from parameter server");
+			// TODO: how to gracefully exit here???
```

Comment:
Just return. It's enough to provide diag_enabled == false.

Reply:
done